nilearn/plotting/surf_plotting.py

```diff
@@ -29,10 +33,416 @@
+}
+
+
+CAMERAS = {
+    "left": {
+        "eye": {"x": -1.7, "y": 0, "z": 0},
+        "up": {"x": 0, "y": 0, "z": 1},
+        "center": {"x": 0, "y": 0, "z": 0},
+    },
+    "right": {
+        "eye": {"x": 1.7, "y": 0, "z": 0},
+        "up": {"x": 0, "y": 0, "z": 1},
+        "center": {"x": 0, "y": 0, "z": 0},
+    },
+    "top": {
+        "eye": {"x": 0, "y": 0, "z": 1.7},
+        "up": {"x": 0, "y": 1, "z": 0},
+        "center": {"x": 0, "y": 0, "z": 0},
+    },
+    "bottom": {
+        "eye": {"x": 0, "y": 0, "z": -1.7},
+        "up": {"x": 0, "y": 1, "z": 0},
+        "center": {"x": 0, "y": 0, "z": 0},
+    },
+    "front": {
+        "eye": {"x": 0, "y": 1.7, "z": 0},
+        "up": {"x": 0, "y": 0, "z": 1},
+        "center": {"x": 0, "y": 0, "z": 0},
+    },
+    "back": {
+        "eye": {"x": 0, "y": -1.7, "z": 0},
+        "up": {"x": 0, "y": 0, "z": 1},
+        "center": {"x": 0, "y": 0, "z": 0},
+    },
+}
+
+
+LAYOUT = {
+    "scene": {f"{dim}axis": AXIS_CONFIG for dim in ("x", "y", "z")},
+    "paper_bgcolor": "#fff",
+    "hovermode": False,
+    "margin": {"l": 0, "r": 0, "b": 0, "t": 0, "pad": 0},
+}
+
+
+def _set_view_plot_surf_plotly(hemi, view):
+    """Helper function for plot_surf with plotly engine.
+
+    This function checks the selected hemisphere and view, and
+    returns the cameras view.
+    """
+    hemis = np.array(['left', 'right'])
+    if hemi not in hemis:
+        raise ValueError(f"'hemi' must be one of {VALID_HEMISPHERES}")
+    if view == 'anterior':
+        cameras_view = 'front'
+    elif view == 'posterior':
+        cameras_view = 'back'
+    elif view == 'dorsal':
+        cameras_view = 'top'
+    elif view == 'ventral':
+        cameras_view = 'bottom'
+    elif view == 'lateral':
+        cameras_view = hemi
+    elif view == 'medial':
+        cameras_view = hemis[hemis != hemi][0]
+    else:
+        raise ValueError(f"view must be one of {VALID_VIEWS}")
+    return cameras_view
+
+
+def _plot_surf_plotly(surf_mesh, surf_map=None, bg_map=None,
+                      hemi='left', view='lateral', cmap=None,
+                      colorbar=False, avg_method='mean',
+                      threshold=None, output_file=None):
+    """Helper function for plot_surf.
+
+    .. versionadded:: 0.8.1
+
+    This function handles surface plotting when the selected
+    engine is plotly.
+    """
+    coords, faces = load_surf_mesh(surf_mesh)
+    x, y, z = coords.T
+    i, j, k = faces.T
+    if cmap is None:
+        cmap = cold_hot
+    if surf_map is not None:
+        surf_map_faces = _compute_surf_map_faces(
+            surf_map, faces, avg_method,
+            coords.shape[0], faces.shape[0]
+        )
+        colors = colorscale(cmap, surf_map_faces, threshold)
+        vertexcolor = _get_vertexcolor(
+            surf_map_faces,
+            colors["cmap"],
+            colors["norm"],
+            colors["abs_threshold"],
+            bg_map=bg_map,
+        )
```

Comment:
Contrary to `matplotlib`, `plotly` can assign colours to vertices instead of faces (and interpolate the former correctly), so I would probably just stick to @jeromedockes' example implementation and not use `_compute_surf_map_faces` here.
This might yield different images depending on what engine you use, but I think this new setup is way more intuitive (in fact I tried to alleviate this in #2790 by adding more `avg_method`s but this PR would do a much better job).

```suggestion
        colors = colorscale(cmap, surf_map, threshold)
        vertexcolor = _get_vertexcolor(
            surf_map,
            colors["cmap"],
            colors["norm"],
            colors["abs_threshold"],
            bg_map=bg_map,
        )
```

Reply:
+1, plot the values of vertices and do not compute values for faces

> This might yield different images depending on what engine you use

yes that's the goal :) -- getting better images than with the matplotlib engine